.ci/e2eFleetServer.groovy

```diff
@@ -0,0 +1,190 @@
+  }
+  stages {
+    stage('Initialize'){
+      agent { label 'ubuntu-20 && immutable' }
+      options { skipDefaultCheckout() }
+      environment {
+        HOME = "${env.WORKSPACE}/${BASE_DIR}"
+      }
+      stages {
+        stage('Check permissions') {
+          steps {
+            checkPermissions()
+            setEnvVar('E2E_BASE_BRANCH', getE2EBaseBranch())
+            sh(label:'Prepare Agent Drop path', script: 'mkdir -p ${AGENT_DROP_PATH}')
+          }
+        }
+        stage('Build Elastic Agent dependencies') {
+          options { skipDefaultCheckout() }
+          parallel {
+            stage('Build Fleet Server') {
+              options { skipDefaultCheckout() }
+              steps {
+                gitCheckout(basedir: BASE_DIR, branch: 'master', githubNotifyFirstTimeContributor: true, repo: "git@github.com:${env.ELASTIC_REPO}.git", credentialsId: env.JOB_GIT_CREDENTIALS)
+                dir("${BASE_DIR}") {
+                  withGoEnv(){
+                    sh(label: 'Build Fleet Server', script: "make release")
+                    sh(label: 'Copy binaries to Agent Drop path', script: 'cp build/distributions/* ${AGENT_DROP_PATH}')
+                  }
+                }
+              }
+            }
+            stage('Build Elastic Agent Dependencies') {
+              options { skipDefaultCheckout() }
+              steps {
+                gitCheckout(basedir: BEATS_BASE_DIR, branch: 'master', githubNotifyFirstTimeContributor: true, repo: "git@github.com:${env.BEATS_ELASTIC_REPO}.git", credentialsId: env.JOB_GIT_CREDENTIALS)
+                dir("${BEATS_BASE_DIR}/x-pack/elastic-agent") {
+                  withGoEnv(){
+                    withEnv(["DEV=true", "SNAPSHOT=true", "PLATFORMS='+all linux/amd64'"]) {
+                      sh(label: 'Build Fleet Server', script: 'mage package')
+                    }
+                  }
+                }
+                dir("${BEATS_BASE_DIR}/x-pack") {
+                  sh(label:'Copy Filebeat binaries to Agent Drop path', script: 'cp filebeat/build/distributions/* ${AGENT_DROP_PATH}')
+                  sh(label:'Copy Heartbeat binaries to Agent Drop path', script: 'cp metricbeat/build/distributions/* ${AGENT_DROP_PATH}')
+                  sh(label:'Copy Metricbeat binaries to Agent Drop path', script: 'cp heartbeat/build/distributions/* ${AGENT_DROP_PATH}')
+                }
+              }
+            }
+          }
+        }
+        stage('Build Elastic Agent including Fleet Server') {
+          options { skipDefaultCheckout() }
+          steps {
+            dir("${BEATS_BASE_DIR}/x-pack/elastic-agent") {
+              withGoEnv(){
+                withEnv(["AGENT_DROP_PATH='${env.AGENT_DROP_PATH}'", "DEV=true", "SNAPSHOT=true", "PLATFORMS='+all linux/amd64'"]) {
+                  sh(label: 'Build Fleet Server', script: 'mage package')
+                }
```

Comment:
NIT: shell step with the entire explicit command
```suggestion
                sh(label: 'Build Fleet Server', script: "AGENT_DROP_PATH='${env.AGENT_DROP_PATH}' DEV=true SNAPSHOT=true PLATFORMS='+all linux/amd64' mage package")
```

Reply:
I considered this style, although I though it was more readable using `withEnv`. OTOH, using your suggestion, I think the commands are more "copy&pasteable", which I like it.